rpc/handlers_test.go

```diff
@@ -1341,3 +1343,132 @@ func TestClassAt(t *testing.T) {
 		assertEqualCairo0Class(t, cairo0Class, class)
 	})
 }
+
+func TestEvents(t *testing.T) {
+	testDB := pebble.NewMemTest()
+	t.Cleanup(func() {
+		require.NoError(t, testDB.Close())
+	})
+	chain := blockchain.New(testDB, utils.GOERLI2, utils.NewNopZapLogger())
+
+	client, closeFn := feeder.NewTestClient(utils.GOERLI2)
+	t.Cleanup(closeFn)
+	gw := adaptfeeder.New(client)
+
+	for i := 0; i < 7; i++ {
+		b, err := gw.BlockByNumber(context.Background(), uint64(i))
+		require.NoError(t, err)
+		s, err := gw.StateUpdate(context.Background(), uint64(i))
+		require.NoError(t, err)
+		require.NoError(t, chain.Store(b, s, nil))
+	}
+
+	handler := rpc.New(chain, nil, utils.MAINNET, utils.NewNopZapLogger())
+	from := utils.HexToFelt(t, "0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7")
+	args := &rpc.EventsArg{
+		EventFilter: &rpc.EventFilter{
+			FromBlock: &rpc.BlockID{Number: 0},
+			ToBlock:   &rpc.BlockID{Latest: true},
+			Address:   from,
+			Keys:      []*felt.Felt{},
+		},
+		ResultPageRequest: &rpc.ResultPageRequest{
+			ChunkSize:         100,
+			ContinuationToken: "",
+		},
+	}
+
+	t.Run("filter non-existent", func(t *testing.T) {
+		t.Run("block number", func(t *testing.T) {
+			args.ToBlock = &rpc.BlockID{Number: 55}
+			_, err := handler.Events(args)
+			require.Equal(t, rpc.ErrBlockNotFound, err)
+		})
+
+		t.Run("block hash", func(t *testing.T) {
+			args.ToBlock = &rpc.BlockID{Hash: new(felt.Felt).SetUint64(55)}
+			_, err := handler.Events(args)
+			require.Equal(t, rpc.ErrBlockNotFound, err)
+		})
+	})
+
+	t.Run("filter with no from_block", func(t *testing.T) {
+		args.FromBlock = nil
+		args.ToBlock = &rpc.BlockID{Latest: true}
+		_, err := handler.Events(args)
+		require.Nil(t, err)
+	})
+
+	t.Run("filter with no to_block", func(t *testing.T) {
+		args.FromBlock = &rpc.BlockID{Number: 0}
+		args.ToBlock = nil
+		_, err := handler.Events(args)
+		require.Nil(t, err)
+	})
+
+	t.Run("filter with no address", func(t *testing.T) {
+		args.ToBlock = &rpc.BlockID{Latest: true}
+		args.Address = nil
+		_, err := handler.Events(args)
+		require.Nil(t, err)
+	})
```

Comment:
I don't see how these tests are meaningful, all they are checking is that there is no error but there are no assert statements on what the actual result should be.

I don't know what the behaviour should be if `fromBlock`/`toBlock`/`contractAddress` is missing from the filter (and the spec isn't any better in specifying what it means).

Reply:
It checks that the arguments have a default value, (ie it is possible to call getEvents with bunch of arguments missing).

in our implementation;

default fromBlock = 0
default toBlock = latest
default contractAddress = nil (which mean what I explained [here](https://github.com/NethermindEth/juno/pull/746#discussion_r1197358092))

Reply:
> default fromBlock = 0
default toBlock = latest

How is this enforced?